### PR TITLE
[13.0][MIG] purchase_order_type

### PR DESCRIPTION
**Pull request by ps-tubtim**
Add company_id

### CODE REVIEW

File: purchase_order_type/tests/test_purchase_order_type.py [hunk 1]
```diff
@@ -65,3 +65,14 @@ def _create_purchase(self, line_products):
...
+    def test_purchase_order_change_company(self):
+        order = self.po_obj.new({"partner_id": self.partner1.id})
+        self.assertEqual(order.order_type, self.type1)
+        order._onchange_company()
+        self.assertFalse(order.order_type)
+
+    def test_purchase_order_type_company_error(self):
+        order = self.po_obj.create({"partner_id": self.partner1.id})
+        self.assertEqual(order.order_type, self.type1)
+        self.assertEqual(order.company_id, self.type1.company_id)
```
Comment (kittiu) [inline purchase_order_type/tests/test_purchase_order_type.py:82]: You should test for error like this,
https://github.com/ecosoft-odoo/sale-workflow/blob/13.0-enh-sale_order_type/sale_order_type/tests/test_sale_order_type.py#L130

**Review by kittiu — APPROVED** [review]

File: purchase_order_type/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+from odoo.addons.purchase.models.purchase import PurchaseOrder as Purchase
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _default_order_type(self):
+        return self.env["purchase.order.type"].search(
+            ["|", ("company_id", "=", False), ("company_id", "=", self.env.company.id)],
```
Comment (ps-tubtim) [inline purchase_order_type/models/purchase_order.py]:
```suggestion
            [("company_id", "in", [False, self.env.company.id])],
```

File: purchase_order_type/models/purchase_order_type.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @api.model
+    def _get_domain_sequence_id(self):
+        seq_type = self.env.ref("purchase.seq_purchase_order")
+        return [
+            ("code", "=", seq_type.code),
+            "|",
+            ("company_id", "=", False),
+            ("company_id", "=", self.env.company.id),
+        ]
```
Comment (ps-tubtim) [inline purchase_order_type/models/purchase_order_type.py]:
```suggestion
        return [
            ("code", "=", seq_type.code),
            ("company_id", "in", [False, self.env.company.id]),
        ]
```

File: purchase_order_type/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+To configure this module, you need to:
+
+* Go to **Purchases > Configuration > Purchase types**
```
Comment (ps-tubtim) [inline purchase_order_type/readme/CONFIGURE.rst:3]:
```suggestion
* Go to **Purchases > Configuration > Purchase Order Types**
```
Please update `CONFIGURE.rst`.

File: purchase_order_type/readme/CONTRIBUTORS.rst [hunk 5]
```diff
@@ -0,0 +1,2 @@
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Pimolnat Suntian <pimolnats@ecosoft.co.th>
```
Comment (ps-tubtim) [inline purchase_order_type/readme/CONTRIBUTORS.rst:2]: Please add you as a contributor.

**Comment by kittiu** [comment]
@ps-tubtim I see all the test are green, not sure why it still red. May be try pushing it again to restart test?

**Comment by kittiu** [comment]
Rebased, hope it can fix. :)

**Comment by kittiu** [comment]
The Travis error is because of module `purchase_order_line_price_history_discount` https://travis-ci.com/github/OCA/purchase-workflow/jobs/454197287#L1609

It error because this PR, `purchase_order_type` was installed before, and already set required field which is unknown to it.

How do you deal with this situation? @OCA/procurement-purchase-maintainers

**Comment by kittiu** [comment]
Fixed like this, is this ok?
https://github.com/OCA/purchase-workflow/compare/f06839db22049a0d055c3da7ac5f26e4ae79a542..745f5b37ee2bf9003233c12f4e3e67a83504537f

**Review by SurachetS — APPROVED** [review]
Functional Tested LGTM.

**Pull request merged**